pandas/core/internals/array_manager.py

```diff
@@ -1145,6 +1145,15 @@ def as_array(
 
         return result
 
+    def as_array_dtype(self):
+        """
+        The dtype of the np.ndarray when you would convert self to a numpy array
+        (i.e. calling ``mgr.as_array()`` or ``df.values``).
+        """
+        if len(self.arrays) == 0:
+            return np.dtype(float)
```

Comment:
can this be explicitly np.float64?  i never know how this will behave on windows or 32bit builds